ReactWindows/ReactNative/Modules/Image/ImageLoaderModule.cs

```diff
@@ -15,8 +48,382 @@ public override string Name
         [ReactMethod]
         public void prefetchImage(string uriString, IPromise promise)
         {
-            // TODO: (#366) Implement prefetch mechanism.
-            promise.Reject("Prefetch is not yet supported.");
+            var image = GetImage(uriString);
+            if (image != null)
+            {
+                ImageReference(uriString, 1, true);
+                promise.Resolve(true);
+                return;
+            }
+
+            RunOnDispatcher(() =>
+            {
+                var imageFailedHandler = new ExceptionRoutedEventHandler(
+                    (sender, args) => OnImageLoadFailedWithPromise(sender, args, promise));
+
+                var imageOpenedHandler = new RoutedEventHandler(
+                    (sender, args) => OnImageLoadedWithPromise(sender, args, false, uriString, promise));
+
+                try
+                {
+                    var uri = new Uri(uriString);
+                    Load(imageOpenedHandler, imageFailedHandler, uri);
+                }
+                catch (Exception ex) when (ex is ArgumentNullException || ex is UriFormatException)
+                {
+                    promise.Reject(ex.Message);
+                }
+            });
+        }
+
+        [ReactMethod]
+        public void getSize(string uriString, IPromise promise)
+        {
+            var image = GetImage(uriString);
+            if (image != null)
+            {
+                var sizes = new JObject()
+                {
+                    new JProperty("width", image.PixelWidth),
+                    new JProperty("height",image.PixelHeight),
+                };
+
+                promise.Resolve(sizes);
+                return;
+            }
+
+            RunOnDispatcher(() =>
+            {
+                var imageFailedHandler = new ExceptionRoutedEventHandler(
+                    (sender, args) => OnImageLoadFailedWithPromise(sender, args, promise));
+
+                var imageOpenedHandler = new RoutedEventHandler(
+                    (sender, args) => OnImageLoadedWithPromise(sender, args, true, uriString, promise));
+
+                try
+                {
+                    var uri = new Uri(uriString);
+                    Load(imageOpenedHandler, imageFailedHandler, uri);
+                }
+                catch (Exception ex) when (ex is ArgumentNullException || ex is UriFormatException)
+                {
+                    promise.Reject(ex.Message);
+                }
+            });
+        }
+
+        internal BitmapImage QueryImage(string uriString)
+        {
+            var image = GetImage(uriString);
+
+            if (image != null)
+            {
+                ImageReference(uriString, 1);
+            }
+
+            return image;
+        }
+
+        internal void FreeReference(string uriString)
```

Comment:
Maybe we can take a more object oriented approach and use something like IDisposable.